config/src/iroha.rs

```diff
@@ -88,48 +87,56 @@ impl ConfigurationProxy {
     /// # Errors
     /// - If the relevant uppermost Iroha config fields were not provided.
     #[allow(clippy::expect_used, clippy::unwrap_in_result)]
-    pub fn finish(&mut self) -> Result<()> {
+    pub fn finish(&mut self) -> Result<(), ConfigError> {
         if let Some(sumeragi_proxy) = &mut self.sumeragi {
             // First, iroha public/private key and sumeragi keypair are interchangeable, but
             // the user is allowed to provide only the former, and keypair is generated automatically,
             // bailing out if key_pair provided in sumeragi no matter its value
             if sumeragi_proxy.key_pair.is_some() {
-                eyre::bail!(ConfigError::ProxyBuildError(
-                    "Sumeragi should not be provided with `key_pair` directly as it is instantiated via Iroha config. Please set the `KEY_PAIR` to `null` or omit them entirely."
-                        .to_owned()))
+                return Err(ConfigError::ProvidedInferredField {
+                    field: "key_pair",
+                    message: "Sumeragi should not be provided with `KEY_PAIR` directly. That value is computed from the other config parameters. Please set the `KEY_PAIR` to `null` or omit entirely."
+                });
             }
             if let (Some(public_key), Some(private_key)) = (&self.public_key, &self.private_key) {
                 sumeragi_proxy.key_pair =
                     Some(KeyPair::new(public_key.clone(), private_key.clone())?);
             } else {
-                eyre::bail!(ConfigError::ProxyBuildError(
-                    "Iroha public and private key not supplied, instantiating `sumeragi` keypair is impossible. Please provide `PRIVATE_KEY` and `PUBLIC_KEY` variables."
-                        .to_owned()
-                ))
+                return Err(ConfigError::MissingField {
+                    field: "PUBLIC_KEY and PRIVATE_KEY",
+                    message: "The sumeragi keypair is not provided in the example configuration. It's done this way to ensure you don't re-use the example keys in production, and know how to generate new keys. Please have a look at \n\nhttps://hyperledger.github.io/iroha-2-docs/guide/configure/keys.html\n\nto learn more.\n\n-----",
```

Comment:
Maybe paraphrasing it to be impersonal would be more polite? Reading that something is fool-proofed is nicer when there's room to assume it's not _you_ who may be the fool in question.